xarray/core/dataarray.py

```diff
@@ -811,12 +812,8 @@ def rename(self, new_name_or_name_dict):
         """
         if utils.is_dict_like(new_name_or_name_dict):
             name_dict = new_name_or_name_dict.copy()
```

Comment:
It occurs to me that the copying here is no longer necessary, since we no longer modify the dictionary.